---
title: "Using International Information In National Single Step Genomic BLUP In Swiss Dairy Cattle"
author: Christian Stricker and Peter von Rohr
date: "2018-05-28"
output: 
  pdf_document:
    pandoc_args: [
      "-V", "classoption=twocolumn"
    ]
header-includes:
- \setlength{\columnsep}{18pt}
bibliography: skeleton.bib
biblio-style: apalike
link-citations: yes
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## Introduction
In Swiss dairy cattle breeding, genomic breeding values are estimated using marker-effect models (MEM) and a Bayesian Regression approach (Bayes $A - C$ and $C_{\pi}$; @Gianola2013). MEM do not provide genomic breeding value estimates for animals that are not genotyped. Furthermore, as phenotypes of non-genotyped animals are not used in the analysis, covariances between genotyped an non-genotyped animals are not contributing to estimates of genomic breeding values. Estimated genomic breeding values are further combined with traditional BLUP-based breeding values via an index procedure. Again in this index approach, genomic breeding values of genotyped individuals do not contribute to estimating animals without SNP-information. The reliability of genomic breeding values is combined with those of traditional BLUP-based breeding values in an ad-hoc manner.

In dairy cattle breeding programs genotypic information is exchanged between different countries, but phenotypic information is not. Hence for foreign sires no phenotypic information or only that from its domestic daughters may be available. Therefore MEM evaluations are usually based on de-regressed MACE breeding values.


## Single Step GBLUP
In dairy cattle, genomic BLUP (GBLUP) is a widely used procedure to estimate genomic breeding values. GBLUP can be parametrised as MEM or as breeding-value-model (BVM) where the sum of all SNP-effects per animal is modeled as a single random component. @LAM2009 showed a single-step GBLUP approach which directly combines information of genotyped animals with animals that do not have any SNP-information. The BVM single-step GBLUP model differs from the traditional BLUP animal model only in the variance-covariance matrix of the random effects 'animal'. For the BVM single-step GBLUP model, the additive genetic relationship matrix used to model the covariance between genotyped relatives is replaced by a genomic relationship matrix. This genomic relationship matrix is often blended with additive genetic relationships. For the non-genotyped animals covariances are linearly estimated based on the additive relationship matrix and the genotyped relatives. The final total variance-covariance matrix again requires a lot of fine-tuning depending on the trait of interest and on the SNP-allele frequencies in the base population (which are unknown). 


## Combining Different Information Sources
As outlined above, the major problem in using information from foreign sires in the national evaluation is that foreign phenotypic information is not available. @VSPG2017 showed how genomic breeding values can be estimated using single-step BVM GBLUP for populations with different sources of information. A second problem is, that simply using deregressed foreign MACE breeding values as phenotypic information together with the national information will result in double counting the national information. Thus, a strategy to combine domestic with foreign information without double counting is required.  In this study, we want to evaluate strategies for combining all available information to come up with a single-step GBLUP approach which is suitable for Swiss dairy cattle breeding populations. 

Further information will be provided on the project website at: https://charlotte-ngs.github.io/SingleStepGBLUP/.

## References
\small
```{r BibRefs, echo=FALSE}
s_bib_file <- "skeleton.bib"
vec_bib <- c(bibentry(
    bibtype = "Article",
    title = "A relationship matrix including full pedigree and genomic information",
    author = c(as.person("A. Legarra"), as.person("I. Aguilar"), as.person("I. Misztal")),
    journal= "Journal of Dairy Science",
    year = 2009,
    key = "LAM2009"),
  bibentry(
    bibtype = "Article",
    author = as.person("D. Gianola"),
    doi = "10.1534/genetics.113.151753",
    journal = "Genetics",
    number = 3,
    pages = "573--596",
    publisher = "Genetics Society of America",
    title = "Priors in Whole-Genome Regression: The Bayesian Alphabet Returns",
    url = "http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3697965/",
    volume = 194,
    year = 2013,
    key = "Gianola2013"),
  bibentry(
    bibtype = "Article",
    author = c(as.person("J. Vandenplas"), 
               as.person("M. Spehar"),
               as.person("K. Potocnik"),
               as.person("N. Gengler"),
               as.person("G. Gorjanc")),
    doi = "10.3168/jds.2016-11733",
    journal = "Journal of Dairy Science",
    title = "National single-step genomic method that integrates multi-national genomic information",
    url = "http://linkinghub.elsevier.com/retrieve/pii/S0022030216308128 http://dx.doi.org/10.3168/jds.2016-11733",
    volume = 100,
    year = 2017,
    number = 1,
    pages = "465--478",
    publisher = "Elsevier",
    key = "VSPG2017"
  ))
rmddochelper::write_bib(pvec_bref = vec_bib)
```

